### PR TITLE
Keystore testing from package

### PR DESCRIPTION
**Pull request by ph**
Create a keys in the keystore and start the beat with a configuration
that use the key, if the key is missing the beat will refuse to start.
This assert that the path used by the packages is correct for the CLI
and the started service.

This is to stop regression like in https://github.com/elastic/beats/pull/11497

I have tested the changes with debian8 and windows box.

### CODE REVIEW

File: roles/test-beat/templates/filebeat.yml.j2 [hunk 1]
```diff
@@ -6,6 +6,8 @@ filebeat.config.modules:
 filebeat.inputs:
 - paths:
   - '{{ filebeat_logs_dir }}/*.log'
+  fields:
```
Comment (ruflin) [inline roles/test-beat/templates/filebeat.yml.j2]: Should this be fields under the input or on the top level?

File: roles/test-beat/templates/metricbeat.yml.j2 [hunk 2]
```diff
@@ -9,6 +9,9 @@ output.file:
   path: '${path.logs}'
   filename: output.json
 
+fields:
```
Comment (ruflin) [inline roles/test-beat/templates/metricbeat.yml.j2]: Sometimes they are on the top level, sometimes not.

File: roles/test-beat/templates/heartbeat.yml.j2 [hunk 3]
```diff
@@ -4,6 +4,8 @@ heartbeat.monitors:
 - type: http
   schedule: '@every 1s'
   urls: ["http://localhost:12345"]
+  fields:
```
Comment (ruflin) [inline roles/test-beat/templates/heartbeat.yml.j2]: Do we support `fields` inside a monitor?

Comment (ph) [reply roles/test-beat/templates/heartbeat.yml.j2]: I will put all of them at the global level.

**Comment by ph** [comment]
@ruflin I've move everything at the top level, I need to run the suite again before merging to make sure I didn't break anything.

File: roles/test-beat/tasks/auditbeat/pre-run.yml [hunk 4]
```diff
@@ -19,3 +19,19 @@
     path: '{{ ansible_user_dir }}/auditbeat_hello.txt'
     state: touch
   when: ansible_system != "Win32NT"
+
+# Verify that keystore works
+- name: Create a keystore
+  shell: auditbeat keystore add ES_PWD --stdin --force
+  args:
+    stdin: "hello"
+  register: auditbeat_keystore
```
Comment (andrewkroh) [inline roles/test-beat/tasks/auditbeat/pre-run.yml]: What do you do with the variables that you register?

**Pull request closed without merging**